### PR TITLE
Stac-13817: upgrade docker agent

### PR DESCRIPTION
**Pull request by vishalxebia**
### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test
- [ ] Integration test
- [ ] E2E/Molecule test

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Did you add release notes describing the changes you made?
- [ ] Yes

### Step 6: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,5 +4,5 @@ reno==2.9.2
 dulwich==0.19.16
 docker==3.7.3
 requests==2.23.0
```
Comment (LouisParkin) [inline requirements.txt:6]: `requests==2.25.0`

File: Dockerfiles/agent/Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # Preparation stage: extract and cleanup #
 ##########################################
 
-FROM debian:buster-slim as extract
+FROM ubuntu:21.04 as extract
```
Comment (LouisParkin) [inline Dockerfiles/agent/Dockerfile:5]: Why Ubuntu (move away from Debian), and not just use Debian Bullseye?
https://hub.docker.com/layers/debian/library/debian/bullseye-slim/images/sha256-fd9bc56e83c76803d355ddbe0a4e18aae08f4ee16d05a6fafb843ee90cc0104c?context=explore

Comment (LouisParkin) [reply Dockerfiles/agent/Dockerfile:5]: `docker pull debian:bullseye-slim`

File: requirements.txt (hunk 1 above)
Comment (LouisParkin) [reply requirements.txt:6]: And please add `urllib3==1.26.5`

File: Dockerfiles/cluster-agent/Dockerfile [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # Preparation stage: layout and chmods #
 ########################################
 
-FROM debian:buster-slim as builder
+FROM ubuntu:20.04 as builder
```
Comment (julianosk) [inline Dockerfiles/cluster-agent/Dockerfile:5]: Pull from StackState artifactory

File: omnibus/config/software/python3.rb [hunk 4]
```diff
@@ -38,6 +39,9 @@
   build do
     ship_license "PSFL"
 
+    # security patches
```
Comment (julianosk) [inline omnibus/config/software/python3.rb:42]: Add a comment to remove this patch once python 3 is updated to 3.8.11, 3.9 or 3.10

File: omnibus/config/software/stackstate-agent-integrations-py2.rb [hunk 5]
```diff
@@ -292,10 +304,10 @@
     # Patch applies to only one file: set it explicitly as a target, no need for -p
     if windows?
       patch :source => "create-regex-at-runtime.patch", :target => "#{python_2_embedded}/Lib/site-packages/yaml/reader.py"
-      patch :source => "jpype_0_7.patch", :target => "#{python_2_embedded}/Lib/site-packages/jaydebeapi/__init__.py"
+#       patch :source => "jpype_0_7.patch", :target => "#{python_2_embedded}/Lib/site-packages/jaydebeapi/__init__.py"
```
Comment (julianosk) [inline omnibus/config/software/stackstate-agent-integrations-py2.rb:307]: Remove these commented lines

File: omnibus/config/software/stackstate-agent-integrations-py3.rb [hunk 6]
```diff
@@ -291,11 +302,11 @@
     end
 
     # Patch applies to only one file: set it explicitly as a target, no need for -p
-    if windows?
-      patch :source => "jpype_0_7.patch", :target => "#{python_3_embedded}/Lib/site-packages/jaydebeapi/__init__.py"
-    else
-      patch :source => "jpype_0_7.patch", :target => "#{install_dir}/embedded/lib/python3.8/site-packages/jaydebeapi/__init__.py"
-    end
+#     if windows?
```
Comment (julianosk) [inline omnibus/config/software/stackstate-agent-integrations-py3.rb:305]: Remove

File: stackstate-deps.json [hunk 7]
```diff
@@ -1,5 +1,5 @@
 {
-  "STACKSTATE_INTEGRATIONS_VERSION": "1.15.0",
+  "STACKSTATE_INTEGRATIONS_VERSION": "stac-13817-upgrade-docker-agent",
```
Comment (julianosk) [inline stackstate-deps.json:2]: Update to master

**Review by julianosk — CHANGES_REQUESTED** [review]

**Review by julianosk — APPROVED** [review]
LGTM

**Pull request merged**